packages/account-v2/src/containers/PersonalDetailsFormWithExample/PersonalDetailsFormWithExample.tsx

```diff
@@ -1,40 +1,45 @@
-import React, { useMemo } from 'react';
+import React, { useEffect, useMemo } from 'react';
 import { Field, FieldProps, useFormikContext } from 'formik';
 import { InferType } from 'yup';
-import { DerivLightNameDobPoiIcon } from '@deriv/quill-icons';
 import { Checkbox, InlineMessage, Text } from '@deriv-com/ui';
 import { DatePicker } from '../../components/DatePicker';
 import { FormInputField } from '../../components/FormFields';
-import { getNameDOBValidationSchema } from '../../utils';
+import { generateRequiredNameDOBFields, getNameDOBValidationSchema } from '../../utils';
 import { validateField } from '../../utils/validation';
+import { SampleImage } from './SampleImage';
 
 type TPersonalDetailsFormWithExampleValues = InferType<ReturnType<typeof getNameDOBValidationSchema>>;
 
 type TPersonalDetailsFormWithExampleProps = {
+    errorStatus: string | null;
     onConfirm?: () => void;
 };
 
-export const PersonalDetailsFormWithExample = ({ onConfirm }: TPersonalDetailsFormWithExampleProps) => {
+export const PersonalDetailsFormWithExample = ({ errorStatus, onConfirm }: TPersonalDetailsFormWithExampleProps) => {
     const formik = useFormikContext<TPersonalDetailsFormWithExampleValues>();
 
     if (!formik) {
         throw new Error('PersonalDetailsFormWithExample must be used within a Formik component');
     }
 
-    const { errors, values } = formik;
+    const { dirty, errors, values } = formik;
+
+    useEffect(() => {
+        if (values.nameDOBConfirmation && onConfirm) {
+            onConfirm();
+        }
+    }, [values, onConfirm]);
+
+    const fieldsToDisplay = generateRequiredNameDOBFields(errorStatus);
+
+    const validationSchema = getNameDOBValidationSchema().pick(fieldsToDisplay);
 
     const isDisabled = useMemo(() => {
```

Comment:
why do we need here useMemo?
we use values and errors from formik, so with every input at least errors object is always new, right? so there is no sense to use useMemo here, it just add complexity to the code.
what do you think?